eng/pipelines/runtime-codeql.yml

```
@@ -28,48 +28,35 @@ variables:
 extends:
   template:  /eng/pipelines/common/templates/pipeline-with-resources.yml
   parameters:
+    isOfficialBuild: true
     stages:
     - stage: Build
       jobs:
-
       #
-      # Build CoreCLR runtime packs
+      # Build all runtime packs to ensure that the SDL tooling can analyze the code in this repo (including all dependencies on all platforms).
       #
       - template: /eng/pipelines/common/platform-matrix.yml
         parameters:
           jobTemplate: /eng/pipelines/common/global-build-job.yml
           buildConfig: release
           platforms:
-          - linux_x64
           - windows_x64
-          jobParameters:
-            buildArgs: -s clr+libs+host+packs -c $(_BuildConfig)
-            nameSuffix: AllSubsets_CoreCLR
-            isOfficialBuild: ${{ variables.isOfficialBuild }}
-            timeoutInMinutes: 360
-            preBuildSteps:
-              - task: CodeQL3000Init@0
-                displayName: Initialize CodeQL (manually-injected)
-            postBuildSteps:
-              - task: CodeQL3000Finalize@0
-                displayName: Finalize CodeQL (manually-injected)
-
-      #
-      # Build Mono runtime packs
-      #
-      - template: /eng/pipelines/common/platform-matrix.yml
-        parameters:
-          jobTemplate: /eng/pipelines/common/global-build-job.yml
-          buildConfig: release
-          runtimeFlavor: mono
-          platforms:
+          - osx_arm64
           - linux_x64
-          - windows_x64
+          - linux_musl_arm64
+          - android_x64
+          - maccatalyst_x64
+          - tvos_arm64
+          - ios_arm64
+          - linux_bionic_x64
+          - browser_wasm
+          - wasi_wasm
           jobParameters:
-            buildArgs: -s mono+libs+host+packs -c $(_BuildConfig)
-            nameSuffix: AllSubsets_Mono
-            isOfficialBuild: ${{ variables.isOfficialBuild }}
-            timeoutInMinutes: 360
+            templatePath: 'templates-official'
+            buildArgs: -c $(_BuildConfig) -restore -build /p:DotNetBuildAllRuntimePacks=true
```

We need a job that builds all libraries packages. Without the `-pack` switch, only the NetCoreAppCurrent TFMs are getting built.

```suggestion
            buildArgs: -c $(_BuildConfig) -restore -build -pack /p:DotNetBuildAllRuntimePacks=true
```